tests/s25Main/campaign/testCampaignLuaFile.cpp

```diff
@@ -0,0 +1,211 @@
+    BOOST_TEST(desc.luaFolder == "<RTTR_GAME>/CAMPAINGS/ROMAN");
+
+    // maps
+    BOOST_TEST(desc.mapNames.size() == static_cast<size_t>(3));
+    BOOST_TEST(desc.mapNames[0] == "dessert0.WLD");
+    BOOST_TEST(desc.mapNames[1] == "dessert1.WLD");
+    BOOST_TEST(desc.mapNames[2] == "dessert2.WLD");
+}
+
+BOOST_AUTO_TEST_CASE(LoadCampaignDescriptionFailsDueToMissingCampaignVariable)
+{
+    rttr::test::TmpFolder tmp;
+    {
+        bnw::ofstream file(tmp.get() / "campaign.lua");
+
+        file << "roemer_campaign ={\
+            version = \"1\",\
+            author = \"Max Meier\"\
+        }";
+    }
+
+    CampaignDescription desc;
+    CampaignDataLoader loader(desc, tmp.get());
+    rttr::test::LogAccessor logAcc;
+    BOOST_TEST(!loader.Load());
+    REQUIRE_LOG_CONTAINS(
+      "Failed to load campaign data!\nReason: Failed to load game data: Campaign table variable missing.");
+}
+
+BOOST_AUTO_TEST_CASE(LoadCampaignDescriptionFailsDueToIncorrectDifficulty)
+{
+    rttr::test::TmpFolder tmp;
+    {
+        bnw::ofstream file(tmp.get() / "campaign.lua");
+
+        file << "campaign ={\
+            version = \"1\",\
+            author = \"Max Meier\",\
+            name = \"Meine Kampagne\",\
+            shortDescription = \"Sehr kurze Beschreibung\",\
+            longDescription = \"Das ist die lange Beschreibung\",\
+            image = \"<RTTR_GAME>/GFX/PICS/WORLD.LBM\",\
+            maxHumanPlayers = 1,\
+            difficulty = \"middle\",\
+            mapFolder = \"<RTTR_GAME>/DATA/MAPS\",\
+            luaFolder = \"<RTTR_GAME>/CAMPAINGS/ROMAN\",\
+            maps = { \"dessert0.WLD\", \"dessert1.WLD\", \"dessert2.WLD\"}\
+        }";
+    }
+
+    CampaignDescription desc;
+    CampaignDataLoader loader(desc, tmp.get());
+    rttr::test::LogAccessor logAcc;
+    BOOST_TEST(!loader.Load());
+    REQUIRE_LOG_CONTAINS("Failed to load campaign data!\nReason: Invalid difficulty: middle");
+}
+
+BOOST_AUTO_TEST_CASE(LoadCampaignDescriptionFailsDueToMissingField)
+{
+    rttr::test::TmpFolder tmp;
+    {
+        bnw::ofstream file(tmp.get() / "campaign.lua");
+
+        file << "campaign ={\
+            version = \"1\",\
+            author = \"Max Meier\",\
+            name = \"Meine Kampagne\",\
+            shortDescription = \"Sehr kurze Beschreibung\",\
+            longDescription = \"Das ist die lange Beschreibung\",\
+            image = \"<RTTR_GAME>/GFX/PICS/WORLD.LBM\",\
+            maxHumanPlayers = 1,\
+            difficulty = \"easy\",\
+            mapFolder = \"<RTTR_GAME>/DATA/MAPS\",\
+            maps = { \"dessert0.WLD\", \"dessert1.WLD\", \"dessert2.WLD\"}\
+        }";
+    }
+
+    CampaignDescription desc;
+    CampaignDataLoader loader(desc, tmp.get());
+    rttr::test::LogAccessor logAcc;
+    BOOST_TEST(!loader.Load());
+    REQUIRE_LOG_CONTAINS(
+      "Failed to load campaign data!\nReason: Failed to load game data: Required field 'luaFolder' not found");
```

Comment:
Maybe just throw a `std::runtime_error` such that this reads:
```suggestion
    REQUIRE_LOG_CONTAINS("Failed to load campaign data!\nReason: Required field 'luaFolder' not found");
```

Reply:
Thats not possible because CheckedLuaTable throws this error internal.